### PR TITLE
Clang 10

### PR DESCRIPTION
**Pull request by achamayou**
Look into #2719, as discussed with @eddyashton, going to break this up into:

- A set of changes that allow CCF to build with 10, but still default to 10 (#2736)
- Playbook changes necessary to upgrade the toolchain and clang format config, to be parked for now

The first one can go in now, the second will wait until after we've gone through mandatory patches coming from Open Enclave.

### CODE REVIEW

**Review by eddyashton — APPROVED** [review]
Does `clang-format-10` cause any formatting changes to our code? If it does, we should choose a point to do a single formatting pass and then start using the new toolchain everywhere.

File: src/node/code_id.h [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace ccf
     std::array<uint8_t, crypto::Sha256Hash::SIZE> data;
 
     CodeDigest() = default;
-    CodeDigest(const CodeDigest& other) : data(other.data) {}
+    CodeDigest(const CodeDigest&) = default;
```
Comment (ashamis) [inline src/node/code_id.h:18]: Doesn't this change the logic here?

File: src/node/node_signature.h [hunk 2]
```diff
@@ -20,11 +20,7 @@ namespace ccf
     /// Hashed nonce created by the node, only used for BFT
     Nonce hashed_nonce;
 
-    NodeSignature(const NodeSignature& ns) :
-      sig(ns.sig),
-      node(ns.node),
-      hashed_nonce(ns.hashed_nonce)
-    {}
+    NodeSignature(const NodeSignature& ns) = default;
```
Comment (ashamis) [inline src/node/node_signature.h:23]: Doesn't this change the logic here?

Comment (achamayou) [reply src/node/node_signature.h:23]: How so?

File: src/node/code_id.h (hunk 1 above)
Comment (achamayou) [reply src/node/code_id.h:18]: No?

File: src/node/node_signature.h (hunk 2 above)
Comment (ashamis) [reply src/node/node_signature.h:23]: sorry my mistake

**Review by ashamis — APPROVED** [review]

**Comment by achamayou** [comment]
@eddyashton I totally missed that, I'll find out now.

**Pull request closed without merging**